### DISCUSSION TITLE
Can cargo be used in pre-build?

### DISCUSSION THREAD

**Discussion by DefiCake** [cross-rs/cross#1461] [Q&A]
For example:

[target.x86_64-unknown-linux-gnu]
pre-build = [
    "cargo --help"
]

This fails with `/bin/sh: 1: eval: cargo: not found` 

Which means we cannot install useful globals, frameworks and other things.

**Comment by Emilgardis** [comment]
No, cargo is not installed in the image. It is only mounted when actually running the image, and the mounted cargo is the cargo on your own filesystem

What is it that you want to install?

**Reply by DefiCake** [reply]
I see. I wanted to follow these installation instructions https://dev.risczero.com/api/zkvm/install

```
cargo install cargo-binstall
cargo binstall cargo-risczero
cargo risczero install
```

Is there any other stage or config where I could do this?

**Reply by NobodyXu** [reply]
FYI, cargo-binstall has a pre-built binary and you can actually use cargo-binstall without cargo:

```bash
curl -L --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/cargo-bins/cargo-binstall/main/install-from-binstall-release.sh | bash
cargo-binstall cargo-risczero
```

Many cargo-plugin tools can be used without cargo, by invoking the binary directly.

**Reply by Emilgardis** [reply]
Seems like its its own rust toolchain installed via rustup toolchain link, so as long as it runs on x86_64-unknown-linux-gnu you should be able to do `cross +toolchain` and follow the suggestions (you may have to set `CROSS_CUSTOM_TOOLCHAIN=1`). Ofc, if it installs other things outside of the toolchain folder it probably wouldn't work.

Basically, my earlier statement was a bit vague, the more correct statement is:

cross mounts the rust toolchain, as specified by the currently active toolchain channel (nightly, beta, stable) + the [platform](https://github.com/cross-rs/cross/blob/085092ca01d6ac9f5a0b612d9e68bcdf1e50f7ee/src/docker/provided_images.rs#L8) the image runs as. toolchain = cargo home and toolchain folders
